packages/core/src/app/shipping/shippingOption/MultiShippingOptionsListItemV2.tsx

```diff
@@ -0,0 +1,50 @@
+import { ShippingOption } from '@bigcommerce/checkout-sdk';
+import React, { FunctionComponent, useCallback, useEffect } from 'react';
+
+import { RadioInput } from '@bigcommerce/checkout/ui';
+
+import { ShopperCurrency } from '../../currency';
+
+interface MultiShippingOptionsListItemV2Props {
+    consignmentId: string;
+    selectedShippingOptionId?: string;
+    shippingOption: ShippingOption;
+    handleSelect: (value: string) => void;
+}
+
+export const MultiShippingOptionsListItemV2: FunctionComponent<
+    MultiShippingOptionsListItemV2Props
+> = ({ consignmentId, selectedShippingOptionId, shippingOption, handleSelect }) => {
+    const label = (
+        <>
+            {`${shippingOption.description} - `}
+            <ShopperCurrency amount={shippingOption.cost} />
+        </>
+    );
+
+    const selectThisOption = useCallback(() => {
+        handleSelect(shippingOption.id);
+    }, [handleSelect, shippingOption.id]);
+
+    useEffect(() => {
+        if (!selectedShippingOptionId && shippingOption.isRecommended) {
+            selectThisOption();
+        }
+    }, [selectedShippingOptionId, shippingOption.isRecommended, selectThisOption]);
```

Comment:
If we move this logic before rendering the list, could that eliminate the need for `useEffect`?

Reply:
If I understand correctly, regardless of where we place this block of code, except the API layer, it will always require the use of `useEffect`.
Because, after creating a new consignment with items and an address, the shipping option may initially be empty, and we need to compare the values and set the recommended one..